File: Problems/LeetCode 75/1768. Merge Strings Alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {

    let minWord = 0;

    let minWordLength = 0;

    if (word1.length <= word2.length){
        minWordLength = word1.length
        minWord = 1;
    } else {
        minWordLength = word2.length
        minWord = 2;
    }

    let word3 = "";

    for (let i = 0; i < minWordLength; i++){
        word3 += word1.charAt(i);
        word3 += word2.charAt(i)
    }

    if (minWord = 1){
        word3 += word2.slice(minWordLength);
    }
    
    if (minWord = 2){
        word3 += word1.slice(minWordLength);
    }

    return word3;
};

File: Problems/LeetCode 75/ArrayString/1768. Merge Strings Alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        minWord = 0
        minWordLength = 0

        if len(word1) <= len(word2):
            minWordLength = len(word1)
            minWord = 1
        else:
            minWordLength = len(word2)
            minWord = 2

        word3 = ""

        for i in range(minWordLength):
            word3 += word1[i]
            word3 += word2[i]

        if minWord == 1:
            word3 += word2[minWordLength:]

    
        if minWord == 2:
            word3 += word1[minWordLength:]

        return word3
